enhancements/network/drop-node-hostname-resolution.md

```diff
@@ -0,0 +1,198 @@
+use of mDNS to provide hostname resolution. This was recently removed and will
+need to be updated regardless of the outcome of this discussion. For the
+purposes of this enhancement, the main thing to consider is that coredns is
+providing hostname resolution and using compute resources on the nodes to do
+so. The specific mechanism it uses is not necessarily important.
+
+### Risks and Mitigations
+
+* There may be a component in OpenShift that relies on node hostname resolution
+  that we haven't found. This proposal is the mitigation for that - we hope
+  that by getting more eyes on the change any such oversights will be
+  caught. Also, this is a fairly self-contained change so if we had to
+  revert it because a problem was found that should not be an issue.
+
+* The cloud platforms tend to get node name resolution for free from the
+  cloud. It's possible a future change could introduce a dependency on that
+  which would break if on-prem platforms are not providing the same
+  functionality. This proposal is also the mitigation for that. If we have
+  general agreement that node name resolution cannot be assumed then future
+  designs will need to take that into account. We also have on-prem ci jobs
+  running that would quickly catch any such problems.
+
+* A deployer may have third-party services in their cluster that depend on
+  node name resolution. It's debatable whether our internal DNS configuration
+  should be considered a public interface, and in such use cases the name
+  resolution could be provided by external DNS. Some baremetal IPI deployments
+  are already doing that.
+
+## Design Details
+
+### Open Questions [optional]
+
+No one I have talked to is aware of any current requirement for node hostname
+resolution, but no one has been able to say that there definitely isn't one
+either. Is there anyone who could give us a definitive answer on this?
+
+### Test Plan
+
+As this would become the default behavior for on-prem IPI deployments, it
+would be covered by existing tests. The full test suite should continue to
+pass after this change is made. This introduces no new behavior that would
+need to be tested.
+
+### Graduation Criteria
+
+This is not strictly a feature and wouldn't go through a graduation process.
+It's more of an RFC to confirm that we haven't missed anything. Once node
+name resolution is removed this will become the supported configuration for
+on-prem IPI deployments. Ideally, the change should be invisible to users,
+so there is no need for a dev or tech preview phase.
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+N/A
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+On upgrade, the coredns static pod will be removed and replaced with an
+instance of dnsmasq running on the host that will only provide the api-int
+record. We will need to resolve port conflicts between coredns and dnsmasq
```

Comment:
I'm not sure if we should handle the corednד/dnsmasq conflict for upgrade/downgrade, 
the dns server (dnsmasq/coreDNS) is used only by the local node,
from what I recall, in MCO upgrade/downgrade path node being rebooted first and then start with the new MCO image (for upgrade: it will include  dnsmasq ,  for downgrade: it should include coredns)